components/builder-web/app/oauth-providers/index.ts

```diff
@@ -63,6 +66,29 @@ export abstract class OAuthProvider {
   }
 }
 
+class ActiveDirectoryProvider extends OAuthProvider {
+  name: string = 'Active Directory';
+
+  constructor(clientID: string, authorizeUrl: string, redirectUrl: string, signupUrl: string, state: string) {
+    super(
+      OAuthProviderType.ActiveDirectory,
+      clientID,
+      authorizeUrl,
+      redirectUrl,
+      signupUrl,
+      true,
+      {
+        client_id: clientID,
+        redirect_uri: redirectUrl,
+        response_type: 'code',
+        state: state,
+        scope: 'openid',
+        nonce: 0 // Ok since we're not using id_token
```

Comment:
Are all of these properties actually required, or does the destination service just ignore the ones it doesn't need? I only ask because I'd like to restrict them to only what's needed.

Reply:
not entirely sure - I won't like, I carbon copied these from all the others